test/specs/components/Menu/MenuItem-test.tsx

```diff
@@ -46,203 +47,36 @@ describe('MenuItem', () => {
     handlesAccessibility(MenuItem, { defaultRootRole: 'presentation', usesWrapperSlot: true })
     handlesAccessibility(MenuItem, { defaultRootRole: 'menuitem', partSelector: 'a' })
 
-    describe('as a default MenuItem', () => {
-      test('root role should be presentation', () => {
-        const menuItemComponent = mountWithProviderAndGetComponent(MenuItem, <MenuItem />)
-        expect(getRenderedAttribute(menuItemComponent, 'role', '')).toBe('presentation')
-      })
-
-      test('anchor role should be menuitem', () => {
-        const menuItemComponent = mountWithProviderAndGetComponent(MenuItem, <MenuItem />)
-        expect(getRenderedAttribute(menuItemComponent, 'role', 'a')).toBe('menuitem')
-      })
+    const accessibility: { name: string; behavior: Accessibility; expectedAnchorRole: string }[] = [
+      { name: 'default', behavior: undefined, expectedAnchorRole: 'menuitem' },
+      {
+        name: 'toolbarButtonBehavior',
+        behavior: toolbarButtonBehavior,
+        expectedAnchorRole: 'button',
+      },
+      { name: 'tabBehavior', behavior: tabBehavior, expectedAnchorRole: 'tab' },
+    ]
+    accessibility.forEach(accessibility => {
+      test(`integration test for ${accessibility.name} behavior`, () => {
+        // accessibility functionality is covered by a combination of behavior tests and `handlesAccessibility()`
+        // this is just an integration smoke test
 
-      test('aria-label should be added to the child anchor', () => {
         const ariaLabel = 'Useful Tool Tip'
-        const menuItemComponent = mountWithProviderAndGetComponent(
-          MenuItem,
-          <MenuItem aria-label={ariaLabel} />,
-        )
-
-        expect(getRenderedAttribute(menuItemComponent, 'aria-label', '')).toBe(undefined)
-        expect(getRenderedAttribute(menuItemComponent, 'aria-label', 'a')).toBe(ariaLabel)
-      })
 
-      test('aria-labelledby should be added to the child anchor', () => {
-        const ariaLabelledByID = 'element-that-labels'
         const menuItemComponent = mountWithProviderAndGetComponent(
           MenuItem,
-          <MenuItem aria-labelledby={ariaLabelledByID} accessibility={toolbarButtonBehavior} />,
-        )
-
-        expect(getRenderedAttribute(menuItemComponent, 'aria-labelledby', '')).toBe(undefined)
-        expect(getRenderedAttribute(menuItemComponent, 'aria-labelledby', 'a')).toBe(
-          ariaLabelledByID,
-        )
-      })
-    })
-
-    describe('as a ToolbarButton', () => {
-      test('root role should be presentation', () => {
-        const menuItemComponent = mountWithProviderAndGetComponent(
-          MenuItem,
-          <MenuItem accessibility={toolbarButtonBehavior} />,
+          <MenuItem disabled aria-label={ariaLabel} accessibility={accessibility.behavior} />,
```

Comment:
It might also worth to check that onClick handler wouldn't be called when `disabled` prop is passed

Reply:
We have a test for `onClick` in `Menu-test.tsx`, I will add a test for this there.